From 8f84a807e551c7cb6927555a434cb2242ecc0814 Mon Sep 17 00:00:00 2001
From: Kexy Biscuit <kexybiscuit@aosc.io>
Date: Wed, 4 Dec 2024 01:06:04 +0800
Subject: [PATCH 129/132] AOSCOS: MIPS: Loongson: Fix constant timer requesting

In cbe16f35bee6 ("genirq: Add IRQF_NO_AUTOEN for request_irq/nmi()"),
IRQF_NO_AUTOEN is required to be set along with IRQF_SHARED. Setting the
flag for Loongson64 constant timer too.

Fixes: cbe16f35bee6 ("genirq: Add IRQF_NO_AUTOEN for request_irq/nmi()")
Fixes: "AOSCOS: MIPS: Loongson: Add constant timer support"
Fixes: "AOSCOS: MIPS: loongson64: fix constant timer build on kernel versions >= v5.7-rc2"
Cc: Mingcong Bai <jeffbai@aosc.io>
Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 arch/mips/loongson64/constant_timer.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/mips/loongson64/constant_timer.c b/arch/mips/loongson64/constant_timer.c
index ca8b2c14602f..5ae649640116 100644
--- a/arch/mips/loongson64/constant_timer.c
+++ b/arch/mips/loongson64/constant_timer.c
@@ -189,7 +189,7 @@ int constant_clockevent_init(void)
 	constant_timer_irq_installed = 1;
 
 	err = request_irq(irq, constant_timer_interrupt,
-		    IRQF_PERCPU | IRQF_TIMER | IRQF_SHARED,
+		    IRQF_PERCPU | IRQF_TIMER | IRQF_SHARED | IRQF_NO_AUTOEN,
 		    "timer", NULL);
 	if (err) {
 		pr_err("loongson64: setup irq for constant_clock_event failed: %d\n", err);
-- 
2.47.1

